go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -181,6 +190,35 @@ func (bsnc *BaseSecondaryNetworkController) DeleteSecondaryNetworkResourceCommon
 			return err
 		}
 
+	case factory.IPAMClaimsType:
+		ipamClaim, ok := obj.(*ipamclaimsapi.IPAMClaim)
+		if !ok {
+			return fmt.Errorf("could not cast obj of type %T to *ipamclaimsapi.IPAMClaim", obj)
+		}
+
+		var switchType string
+		switch bsnc.NetInfo.TopologyType() {
+		case types.Layer2Topology:
+			switchType = types.OVNLayer2Switch
+		case types.LocalnetTopology:
+			switchType = types.OVNLocalnetSwitch
+		default:
+			klog.Errorf("Persistent IPs not supported for topology %s", bsnc.NetInfo.TopologyType())
+			return nil
+		}
```

Comment:
I don't think we need this check. the IPAMClaim retry framework should not be started it this network is not configured for perssitent IPs.

Reply:
Or just return the error back I guess. Not an scenario we should be overly concerned about I think.